### PR TITLE
Add fast copy and rename demo page

### PR DESCRIPTION
**Pull request by zoready**
- Add fast copy to relevant code blocks 
- Change page URL from /quickstart to /demo
- Create redirect from /quickstart to /demo 
- Update interla links to /demo instead of /quickstart

### CODE REVIEW

**Review by aglover — CHANGES_REQUESTED** [review]
I think there are more references to quickstart throughout the code base:

```
ajglover@Andrews-MacBook-Pro docs % find . -iname '*.mdx' | xargs egrep 'quickstart'
./wip-content/deploy-with-docker.mdx:curl -L -o compose.yml "https://readyset.io/quickstart/compose.yml"
./wip-content/deploy-with-docker.mdx:curl -L -o compose.yml "https://readyset.io/quickstart/compose.yml"
./pages/index.mdx:        <Link href={"/quickstart"} className="h-full">
./pages/quickstart.mdx:  "https://readyset.io/quickstart/compose.postgres.yml" \
./pages/quickstart.mdx:curl -L -s "https://readyset.io/quickstart/imdb-postgres.sql" \
./pages/quickstart.mdx:  "https://readyset.io/quickstart/compose.mysql.yml"    \
./pages/quickstart.mdx:curl -L -s "https://readyset.io/quickstart/imdb-mysql.sql" \
./pages/demos/caching-slow-queries-imdb.mdx:- A ReadySet instance connected to a MySQL or Postgres database with Grafana (we recommend that you use the [ReadySet Demo](/quickstart) to set this up)
./pages/demos/caching-slow-queries-imdb.mdx:curl -L -O "https://readyset.io/quickstart/imdb-mysql.sql"
./pages/demos/caching-slow-queries-imdb.mdx:curl -L -O "https://readyset.io/quickstart/imdb-postgres.sql"
./pages/demos/caching-slow-queries-imdb.mdx:**Note:** if you set up ReadySet via the [ReadySet Demo instructions](/quickstart) (recommended), then you can open Grafana by going to `localhost:4000` in your browser.
./pages/get-started/cache/profiling-queries.mdx:curl -L -O "https://readyset.io/quickstart/metrics.py"
./pages/get-started/cache/profiling-queries.mdx:curl -L -O "https://readyset.io/quickstart/requirements.txt"
```

File: next.config.js
Comment (aglover) [inline next.config.js:1]: Oh I see - this is a mapping that handles automatically quickstart -> demo?

**Review by aglover — APPROVED** [review]
@zoready I think I was overlooking the mapping in _meta.json - I think after clicking around in the preview me might be ok.

File: next.config.js
Comment (zoready) [reply next.config.js:1]: yup! since we've linked it a lot externally, I didn't want it to get fully broken for people

**Pull request merged**